pkg/generator/generator.go

```diff
@@ -212,30 +205,15 @@ func (g Generator) GenerateRawConfig(ctx context.Context, installation, app stri
 	return configmap, secret, nil
 }
 
-func (g Generator) GenerateConfig(ctx context.Context, installation, app, namespace, ref string) (configmap *corev1.ConfigMap, secret *corev1.Secret, err error) {
-	cm, s, err := g.GenerateRawConfig(ctx, installation, app)
+// GenerateConfig generates ConfigMap and Secret for a given App. The generated
+// CM and Secret metadata are configured with the provided value.
+func (g Generator) GenerateConfig(ctx context.Context, app string, meta metav1.ObjectMeta) (*corev1.ConfigMap, *corev1.Secret, error) {
```

Comment:
I decided that generating name and labels here doesn't really work. This will differ between cmd/operator mode and isn't really generic so doesn't fit `pkg` package. Instead we are now apssing `metav1.ObjectMeta` directly.